datafusion/expr/src/expr.rs

```diff
@@ -452,6 +452,13 @@ impl Expr {
             nulls_first,
         }
     }
+
+    pub fn try_into_col(&self) -> Result<Column> {
+        match self {
+            Expr::Column(it) => Ok(it.clone()),
+            _ => plan_err!("Could not coerce into Column!"),
```

Comment:
it might help to include `self` in this error so it is clear what expr is causing issues

like:

```suggestion
            _ => plan_err!("Could not coerce '{}' into Column", self),
```